datafusion/expr/src/expr.rs

```diff
@@ -311,6 +311,10 @@ pub enum Expr {
     ///
     /// This expr has to be resolved to a list of columns before translating logical
     /// plan into physical plan.
+    #[deprecated(
+        since = "46.0.0",
+        note = "A wildcard needs to be resolved to concrete expressions when constructing the logical plan"
```

Comment:
I found a ticket that tracks this work:
- https://github.com/apache/datafusion/issues/7765

```suggestion
        note = "A wildcard needs to be resolved to concrete expressions when constructing the logical plan. See https://github.com/apache/datafusion/issues/7765"
```